### ISSUE TITLE
Transform ansi colour mappings into faces? 

### ISSUE THREAD

**Issue by appetrosyan** [brotzeit/rustic#401]
Currently if one wants to customize the appearance of cargo output, we are limited to colour name mappings. While it's fine in most cases, it would be nice if it were possible to parse out the escaped characters and re-map them to faces, just like what is done with the compilation errors. 

I'm willing to take this up as a task, to contribute to rustic. Will you accept a PR?

**Comment by brotzeit** [comment]
I would definitely accept a PR. But maybe it would be better if this feature would be provided by xterm-color so users could also make use of it for other use cases than rustic ?

**Comment by appetrosyan** [comment]
Good thinking!

**Comment by brotzeit** [comment]
In case the package isn't maintained anymore you can also merge it here.

**Comment by appetrosyan** [comment]
Yes. After a bit of thinking I could just fork it and add the requisite features too.

**Comment by appetrosyan** [comment]
OK, after a bit more thinking I think this still needs to be done on the Rustic side. 

The `rustic` flavour of compilation mode already has some parsing and font-lock, so adding font-locking to `ansi-term` is still not going to give the desired end result. 

We **could** use `xterm-color` text properties to aid in parsing to do proper font-lock, and this could be more efficient than the regex-based parsing of font-lock, but we don't need to modify `xterm-color` for that.

**Comment by appetrosyan** [comment]
No problem, I was a little indisposed myself. 

So there are two ways of processing the compilation messages. 

One is to go full ham and use font lock for everything. It could be slow on some machines, but cargo lines are both short and uniform. The added benefit is that we can turn the names of crates that are being compiled into hyperlinks to the docs of the crate Mong other things. 

A more lightweight approach would be to extend the a-list of text properties that get mapped by ansi-term. Crude but effective (and fast).

**Comment by brotzeit** [comment]
> A more lightweight approach would be to extend the a-list of text properties that get mapped by ansi-term. Crude but effective (and fast).

Ok, if you think it's worth the effort. Let's give it a try.

**Comment by appetrosyan** [comment]
I have to apologise for the delay. I haven't even started work… I expect I'll have a PR ready by the end of July.

**Comment by brotzeit** [comment]
No problem, do it when you find the time and motivation.

**Comment by ed9w2in6** [comment]
Hi, I do not like the colour scheme as well, what I did is that I removed the 2 relevant `setq-local` lines in `rustic-compile.el` and it looked as what I have expected (the default in xterm actually looks nice it bright or dark themes). 

Maybe it is better to just remove the said 2 lines and customisation options and add documentation that defers customisation to `xterm-color.el`.

Or at least set `rustic-ansi-faces` to respect the colours of `ansi-color-<name>` faces.

**Comment by appetrosyan** [comment]
After having around three starts on this I found that the best way to circumvent this problem is to use [`fancy-compilation`](https://github.com/emacsmirror/fancy-compilation) mode. 

It doesn't exactly do what I want it to do with `rustic`, but it fixes the problem of @ed9w2in6. With @ed9w2in6 permission I'd like to close this issue.

**Comment by ed9w2in6** [comment]
Let us keep this open because what we did is just bypassing the issue, BUT the issue is still there.
Whether this is urgent to fix is another discussion.

To clarify my fix, I am referring to these 2 lines:
https://github.com/brotzeit/rustic/blob/39423d1cf4fa054c36bf9577356451f4c06ee148/rustic-compile.el#L187-L188

Literally just commenting them out can be much better.  I do not thing rustic need custom ansi color configuration.
And of course, an even better fix is to rewrite using the Emacs built-in `ansi-color` package.
